### PR TITLE
Added slide expiration date

### PR DESCRIPTION
**Pull request by alfonsolr09**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Sometimes we forget about expired slides on our store. This will help to prevent unwanted old slider banners.
| Type?         | new feature 
| BC breaks?    |  no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/12991
| How to test?  | I copied the pr from @PrestaShark , since he didn't

https://github.com/PrestaShop/ps_imageslider/pull/42#issuecomment-606477725,    #40, #41, #42. 

See this https://github.com/PrestaShop/ps_imageslider/pull/42#issuecomment-606477725

### CODE REVIEW

**Comment by PierreRambaud** [comment]
Hey I think it's a good idea, but could you please create an issue on https://github.com/PrestaShop/PrestaShop/issues/new/choose
IN order to be sure the behavior is ok :)

File: views/templates/hook/list.tpl [hunk 1]
```diff
@@ -45,6 +45,7 @@
 						<div class="col-md-8">
 							<h4 class="pull-left">
 								#{$slide.id_slide} - {$slide.title}
+								{if $slide.expiration|date_format:'%Y-%m-%d' >= $smarty.now|date_format:'%Y-%m-%d' }<span style="color:green">{$slide.expiration}</span>{else}<span style="color:red">{$slide.expiration}</span>{/if}
```
Comment (kpodemski) [inline views/templates/hook/list.tpl:48]: Please use `callback` and leave view without this kind of logic

File: views/templates/hook/slider.tpl [hunk 2]
```diff
@@ -27,17 +27,19 @@
...
-            <img src="{$slide.image_url}" alt="{$slide.legend|escape}" />
-            {if $slide.title || $slide.description }
-              <span class="caption">
-                <h2>{$slide.title}</h2>
-                <div>{$slide.description nofilter}</div>
-              </span>
-            {/if}
-          </a>
-        </li>
+        {if $slide.active AND $slide.expiration|date_format:'%Y-%m-%d' >= $smarty.now|date_format:'%Y-%m-%d'}
```
Comment (kpodemski) [inline views/templates/hook/slider.tpl:30]: Please move this logic into module core

**Review by kpodemski — CHANGES_REQUESTED** [review]
Hey! Great idea.

Few things:
- it would be better to have possibility to use hours and minutes too, full date time
- make sure to add proper field to `renderForm` as we cannot edit `expiration` right now
- make sure to move logic outside `view`

**Comment by alfonsolr09** [comment]
> ¡Oye! Gran idea.
> 
> Pocas cosas:
> 
> * Sería mejor tener la posibilidad de usar horas y minutos también, fecha completa
> * asegúrese de agregar el campo adecuado, `renderForm`ya que no podemos editar en `expiration`este momento
> * asegúrate de mover la lógica afuera `view`

you can do it

**Comment by alfonsolr09** [comment]
@kpodemski I do not know how to do that, can you do it?

**Comment by kpodemski** [comment]
@alfonsolr09 I don't have time now, but I'll bookmark this PR so I can get back to it later

**Comment by alfonsolr09** [comment]
hello, could you do it? @kpodemski

**Comment by PierreRambaud** [comment]
ping @kpodemski are you able to do it? :D

**Comment by kpodemski** [comment]
Hello @alfonsolr09 

We can think about this in the future versions of the `ps_imageslider` module. Right now, the core team doesn't have time to finish this PR.

**Comment by PierreRambaud** [comment]
> @PierreRambaud no, this module needs to be refactored anyway :(
> 
> Btw. if you're here, give me access to this repo, please heart

Done

**Comment by PierreRambaud** [comment]
Hi @alfonsolr09 ,

Since there is no activity for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**